components/suggest/src/db.rs

```diff
@@ -128,63 +128,48 @@ impl<'a> SuggestDao<'a> {
 
     /// Fetches suggestions that match the given query from the database.
     pub fn fetch_suggestions(&self, query: &SuggestionQuery) -> Result<Vec<Suggestion>> {
-        if let Some(suggestion) = self.fetch_yelp_suggestion(query)? {
-            return Ok(vec![suggestion]);
-        }
-
-        let keyword_lowercased = &query.keyword.to_lowercase();
-        let (keyword_prefix, keyword_suffix) = split_keyword(keyword_lowercased);
-        let suggestions_limit = query.limit.unwrap_or(-1);
-
-        let (mut statement, params) = if query
-            .providers
+        let suggestions_limit = query.limit.unwrap_or(i32::MAX);
```

Comment:
Actually, I don't think you need to unwrap here—you can pass along `query` as-is to all the `fetch_*_suggestions()` functions, unwrap-or-default in those, and use an `if-let` when you truncate.

```suggestion
```